Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -503,6 +537,155 @@ int CLuaEngineDefs::EngineImportTXD(lua_State* luaVM)
     return 1;
 }
 
+CClientIMG* CLuaEngineDefs::EngineLoadIMG(lua_State* const luaVM, std::string strFilePath)
+{
+    CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+    if (!pLuaMain)
+        return false;
+
+    // Get the resource we belong to
+    CResource* pResource = pLuaMain->GetResource();
+    if (!pResource)
+        return false;
+
+    std::string strFullPath;
+
+    if (CResourceManager::ParseResourcePathInput(strFilePath, pResource, &strFullPath))
+    {
+        // Grab the resource root entity
+        CClientEntity* pRoot = pResource->GetResourceIMGRoot();
+        // Create the img handle
+        CClientIMG* pImg = new CClientIMG(m_pManager, INVALID_ELEMENT_ID);
+
+        // Attempt loading the file
+        if (pImg->Load(std::move(strFullPath)))
+        {
+            // Success. Make it a child of the resource img root
+            pImg->SetParent(pRoot);
+
+            return pImg;
+        }
+        else
+        {
+            delete pImg;
+            throw std::invalid_argument("Error loading IMG");
+        }
+    }
+
+    throw std::invalid_argument("Bad file path");
+}
+
+bool CLuaEngineDefs::EngineAddImage(CClientIMG* pIMG)
+{
+    return pIMG->StreamEnable();
+}
+
+bool CLuaEngineDefs::EngineRemoveImage(CClientIMG* pIMG)
+{
+    return pIMG->StreamDisable();
+}
+
+uint CLuaEngineDefs::EngineImageGetFilesCount(CClientIMG* pIMG)
+{
+    return pIMG->GetFilesCount();
+}
+
+std::vector<std::string> CLuaEngineDefs::EngineImageGetFileList(CClientIMG* pIMG)
+{
+    uint uiFilesCount = pIMG->GetFilesCount();
+    std::vector<std::string> aOutput;
+    aOutput.reserve(uiFilesCount);
+
+    for (unsigned int i = 0; i < uiFilesCount; i++)
+        aOutput.push_back(pIMG->GetFileInfo(i)->szFileName);
+
+    return aOutput;
+}
+
+std::string CLuaEngineDefs::EngineImageGetFile(CClientIMG* pIMG, std::variant<std::string, uint> file)
+{
+    uint  uiFileID = -1;
+    uint* pFileID = std::get_if<uint>(&file);
+
+    if (pFileID)
+        uiFileID = *pFileID - 1;
+    else
+        uiFileID = pIMG->GetFileID(std::get<std::string>(file));
+
+    if (uiFileID == -1)
+        throw std::invalid_argument("File not found");
+
+    std::string strBuffer;
+    long    lBytesRead = pIMG->GetFile(uiFileID, strBuffer);
+
+    if (lBytesRead >= 0)
+        return strBuffer;
+    else if (lBytesRead == -2)
+        throw std::invalid_argument("Out of memory");
+    else
+        throw std::invalid_argument("File not found");
+}
+
+bool CLuaEngineDefs::EngineImageLinkDFF(CClientIMG* pIMG, std::variant<std::string, uint> file, uint uiModelID)
+{
+    if (uiModelID >= 20000)
+        throw std::invalid_argument("Expected model ID in range [0-19999] at argument 3");
+
+    uint  uiFileID = -1;
+    uint* pFileID = std::get_if<uint>(&file);
+
+    if (pFileID)
+        uiFileID = *pFileID - 1;
+    else
+        uiFileID = pIMG->GetFileID(std::get<std::string>(file));
```

Comment:
Now that you've used this function more than once, I'd say just make a static function for it:
```cpp
static uint ResolveFileID(const std::variant<std::string, uint>& variant)
{
    if (std::holds_alternative<std::string>)
        return pIMG->GetFileID(std::get<std::string>(file));
    return std::get<uint>(file);
}
```
Put it above the img engine functions.